### PR TITLE
Bug 1151337-Perfherder graphs should use OptionCollection model

### PR DESCRIPTION
**Pull request by git-srinivas**
Replaced http.get call with option collection model service call in perf module

[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder-ui/468)

### CODE REVIEW

File: webapp/app/perf.html [hunk 1]
```diff
@@ -36,6 +36,9 @@
   <script src="vendor/mousetrap.min.js"></script>
   <script src="js/treeherder.js"></script>
   <script src="js/providers.js"></script>
+  <script src="js/models/option_collection.js"></script>
+  <script src="js/services/main.js"></script>
+  <script src="js/services/log.js"></script>
```
Comment (wlach) [inline webapp/app/perf.html]: I don't understand why you're including the last two lines.

File: webapp/app/js/perf.js [hunk 2]
```diff
@@ -650,13 +650,16 @@ perf.controller('PerfCtrl', [ '$state', '$stateParams', '$scope', '$rootScope',
...
-        response.data.forEach(function(dict) {
-          optionCollectionMap[dict.option_collection_hash] =
-            dict.options.map(function(option) {
-              return option.name; }).join(" ");
-        });
+    ThOptionCollectionModel.get_list()
+      .success(function(optCollectionData) {
+                 // gather the string representations of option collections
+                 
+                 _.each(optCollectionData, function(optColl) {
```
Comment (wlach) [inline webapp/app/js/perf.js]: The indentation on these lines isn't correct -- it's too deep and inconsistent with the rest of the file. Please follow the style of the original function.

**Comment by wlach** [comment]
This looks ok except for a few stray imports and indentation problems. Does it seem to work ok in your testing? If so, please fix the few issues I mention, amend the commit and push to the fork-- then we can get this merged.

Also, in the future please file review requests for these as outlined in http://ateam-bootcamp.readthedocs.org/en/latest/guide/development_process.html#git-and-github

If you follow the process there it is likely your patch will be looked into sooner.

File: webapp/app/perf.html (hunk 1 above)
Comment (git-srinivas) [inline webapp/app/perf.html]: OptionCollection model uses  thUrl service  which is in main.js  and thUrl service is using  ThLog, it is in log.js. They are the services being used by option collection model.

File: webapp/app/js/perf.js (hunk 2 above)
Comment (git-srinivas) [inline webapp/app/js/perf.js]: Ok lach, i will correct the indentation :)

**Pull request merged**